src/js_results.hpp

```diff
@@ -101,8 +101,11 @@ typename T::Object ResultsClass<T>::create_instance(ContextType ctx, realm::Resu
 }
 
 template<typename T>
-typename T::Object ResultsClass<T>::create_instance(ContextType ctx, SharedRealm realm, const ObjectSchema &object_schema) {
-    auto table = ObjectStore::table_for_object_type(realm->read_group(), object_schema.name);
+typename T::Object ResultsClass<T>::create_instance(ContextType ctx, SharedRealm realm, const std::string &object_type) {
```

Comment:
Yeah we should definitely ship this fix. Was just wondering if there was a more general fix in object store since we have hit this multiple times now.